### PR TITLE
Add OAuth login

### PR DESCRIPTION
**Pull request by helicopter88**
Add login through Google, Github and Facebook
The secrets and keys for each provider are stored inside the vault file, so they're only available in a production environment
TODO:
- Add more providers

### CODE REVIEW

**Comment by jamesreaver** [comment]
Frontend updated to show the OAuth login buttons.

File: src/klee_web/urls.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
     url(r'^manage/', include('control_panel.urls', namespace="control_panel")),
     url(r'^accounts/login/$', auth.views.login,
         {'template_name': 'control_panel/login.html'}),
+    url(r'^soc/',  include('rest_framework_social_oauth2.urls'))
```
Comment (andronat) [inline src/klee_web/urls.py:14]: Who is using this url?

**Comment by andronat** [comment]
This needs rebasing

**Comment by helicopter88** [comment]
Rebased, working

**Pull request merged**